alica_standard_library/src/Conditions.cpp

```diff
@@ -69,4 +69,35 @@ bool AllChildFailure(const alica::Blackboard* input, const alica::RunningPlan* r
     return true;
 }
 
+bool IsAnyChildStatusSuccess(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    return rp->isAnyChildStatus(alica::PlanStatus::Success);
+}
+
+bool AlwaysTrueCond(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    return true;
+}
+
+bool DefaultCondition(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    return false;
+}
+
+bool IsAnyChildStatus(const alica::Blackboard* input, const alica::RunningPlan* rp, const alica::Blackboard* gb)
+{
+    alica::LockedBlackboardRO bb(*input);
+    return rp->isAnyChildStatus(bb.get<alica::PlanStatus>("childStatus"));
```

Comment:
I think we can just have functions for each status (one for success, one for failure)
Don't these already exist though?  with AnyChildSuccess and AnyChildFailure?  what other status is there that we use?

Reply:
There is a difference between AnyChildFailure and IsAnyChildStatusFailed, because AnyChildFailure only checks if a behaviour has failed or if a plan is in a failure state, IsAnyChildStatusFailed also considers plans as failed that have been aborted, plans that should be redone but could not be repaired, failures that should be propagated to the parent with the propagation rule and also plans where the allocation failed

Reply:
@veerajsk Do you think this has general use?

Reply:
At least the FailureHandling.continueOnFailure test in test_continue_on_fail.cpp will fail when we replace isAnyChildStatusFailed with AnyChildFailure. I can return rp->isAnyChildStatus(alica::PlanStatus::Failed) instead of false, to make the test pass and have the additional failure cases covered.

There is also a difference between AnyChildSuccess and IsAnyChildStatusSuccess which fails another test

Reply:
Yeah it's fine not to change the conditions the tests use, I just don't want to add conditions to the standard library unless we are confident in their use